File: app/src/main/java/com/numero/sojodia/manager/NotificationManager.java
package com.numero.sojodia.manager;

import android.app.Notification;
import android.app.NotificationChannel;
import android.content.Context;
import android.content.ContextWrapper;
import android.os.Build;
import android.support.v4.app.NotificationCompat;
import android.support.v4.app.NotificationManagerCompat;

import com.numero.sojodia.R;

public class NotificationManager extends ContextWrapper {

    public final static int UPDATE_NOTIFICATION_ID = 1;

    private static final String CHANNEL_ID = "BUS_DIA_CHANNEL_ID";

    public NotificationManager(Context context) {
        super(context);

        initNotificationChannel();
    }

    private void initNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(CHANNEL_ID, getString(R.string.notification_channel_name), android.app.NotificationManager.IMPORTANCE_LOW);
            channel.setLockscreenVisibility(Notification.VISIBILITY_PUBLIC);

            android.app.NotificationManager manager = (android.app.NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
            if (manager != null) {
                manager.createNotificationChannel(channel);
            }
        }
    }

    public void showNotification() {
        Notification notification = buildNotification();
        notification.flags = Notification.FLAG_ONGOING_EVENT;

        NotificationManagerCompat manager = NotificationManagerCompat.from(getBaseContext());
        manager.notify(UPDATE_NOTIFICATION_ID, notification);
    }

    public void cancelNotification() {
        NotificationManagerCompat manager = NotificationManagerCompat.from(getBaseContext());
        manager.cancel(UPDATE_NOTIFICATION_ID);
    }

    private Notification buildNotification() {
        NotificationCompat.Builder builder = new NotificationCompat.Builder(getApplicationContext(), CHANNEL_ID);
        builder.setSmallIcon(R.drawable.ic_notification);

        builder.setContentTitle(getString(R.string.notification_update_title));
        builder.setContentText(getString(R.string.notification_update_text));
        builder.setProgress(0, 0, true);

        return builder.build();
    }
}

File: app/src/main/kotlin/com/numero/sojodia/manager/NotificationManager.kt
package com.numero.sojodia.manager

import android.app.Notification
import android.app.NotificationChannel
import android.content.Context
import android.content.ContextWrapper
import android.os.Build
import android.support.annotation.RequiresApi
import android.support.v4.app.NotificationCompat
import android.support.v4.app.NotificationManagerCompat

import com.numero.sojodia.R

class NotificationManager(context: Context) : ContextWrapper(context) {

    init {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            initNotificationChannel()
        }
    }

    @RequiresApi(Build.VERSION_CODES.O)
    private fun initNotificationChannel() {
        val channel = NotificationChannel(CHANNEL_ID, getString(R.string.notification_channel_name), android.app.NotificationManager.IMPORTANCE_LOW).apply {
            lockscreenVisibility = Notification.VISIBILITY_PUBLIC
        }

        val manager = getSystemService(Context.NOTIFICATION_SERVICE) as android.app.NotificationManager
        manager.createNotificationChannel(channel)
    }

    fun showNotification() {
        NotificationManagerCompat.from(baseContext).notify(UPDATE_NOTIFICATION_ID, buildNotification().apply {
            flags = Notification.FLAG_ONGOING_EVENT
        })
    }

    fun cancelNotification() {
        NotificationManagerCompat.from(baseContext).cancel(UPDATE_NOTIFICATION_ID)
    }

    private fun buildNotification(): Notification {
        return NotificationCompat.Builder(applicationContext, CHANNEL_ID).apply {
            setSmallIcon(R.drawable.ic_notification)
            setContentTitle(getString(R.string.notification_update_title))
            setContentText(getString(R.string.notification_update_text))
            setProgress(0, 0, true)
        }.build()
    }

    companion object {

        const val UPDATE_NOTIFICATION_ID = 1

        private const val CHANNEL_ID = "BUS_DIA_CHANNEL_ID"
    }
}
